### PR TITLE
4360 hc compatible holofuel

### PR DESCRIPTION
**Pull request by JettTech**
- wraps routes in logical wrapper to discern between hpos and holochain env - to be compatible with holochain env
- updates ws url setting in holochainClient to also work with hc environment

### CODE REVIEW

File: src/holochainClient.js [hunk 1]
```diff
@@ -113,15 +113,20 @@ let holochainClient
...
   try {
-    let url = process.env.NODE_ENV === 'production' ? ('wss://' + window.location.hostname + '/api/v1/ws/') : process.env.REACT_APP_DNA_INTERFACE_URL
-    // Construct url with query param X-Hpos-Admin-Signature = signature
-    const urlObj = new URL(url)
-    const params = new URLSearchParams(urlObj.search.slice(1))
-    params.append('X-Hpos-Admin-Signature', await signPayload('get', urlObj.pathname))
-    params.sort()
-    urlObj.search = params.toString()
-    url = urlObj.toString()
+    if (process.env.REACT_APP_HOLOFUEL_APP === 'true') {
```
Comment (zo-el) [inline src/holochainClient.js]: Can we rename this `REACT_APP_HOLOFUEL_APP` env var?
Because its use is unclear.

**Review by zo-el — CHANGES_REQUESTED** [review]

File: src/holochainClient.js (hunk 1 above)
Comment (JettTech) [reply src/holochainClient.js]: :+1: Good point, this name is a bit outdated now that HoloFuel is bundled within hp-admin when served through the HPOS, and is really only a pure "Holofuel" app whenever run within a Holochain environment. 

Considering the current distinction, I purpose renaming it to : `REACT_APP_HOLOCHAIN_APP`.  Thoughts?

Comment (zo-el) [reply src/holochainClient.js]: Its is more like `holofuel in holochain mode`.

**Review by zo-el — APPROVED** [review]

File: src/holochainClient.js (hunk 1 above)
Comment (JettTech) [reply src/holochainClient.js]: Ok decided on `REACT_APP_HOLOFUEL_AS_HOLOCHAIN_APP` and pushed that up in the last update.

File: src/holofuel/HFRouter.js [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
-import { Switch, Redirect } from 'react-router-dom'
-
+import { Route, Switch, Redirect } from 'react-router-dom'
 import Home from 'holofuel/pages/Home'
 import Inbox from 'holofuel/pages/Inbox'
 import TransactionHistory from 'holofuel/pages/TransactionHistory'
 import CreateOfferRequest from 'holofuel/pages/CreateOfferRequest'
 
+function HFRoute (props) {
+  if (process.env.REACT_APP_HOLOFUEL_AS_HOLOCHAIN_APP === 'true') {
```
Comment (robbiecarlton) [inline src/holofuel/HFRouter.js]: This should use `REACT_APP_HOLOFUEL_APP` (see overall review comment)

File: src/holochainClient.js [hunk 3]
```diff
@@ -113,15 +113,20 @@ let holochainClient
...
   try {
-    let url = process.env.NODE_ENV === 'production' ? ('wss://' + window.location.hostname + '/api/v1/ws/') : process.env.REACT_APP_DNA_INTERFACE_URL
-    // Construct url with query param X-Hpos-Admin-Signature = signature
-    const urlObj = new URL(url)
-    const params = new URLSearchParams(urlObj.search.slice(1))
-    params.append('X-Hpos-Admin-Signature', await signPayload('get', urlObj.pathname))
-    params.sort()
-    urlObj.search = params.toString()
-    url = urlObj.toString()
+    if (process.env.REACT_APP_HOLOFUEL_AS_HOLOCHAIN_APP === 'true') {
```
Comment (robbiecarlton) [inline src/holochainClient.js]: This should use `REACT_APP_RAW_HOLOCHAIN` (see overall review comment)

File: src/holofuel/components/SideMenu/SideMenu.js [hunk 4]
```diff
@@ -56,7 +56,7 @@ export default function SideMenu ({
               History
             </Link>
           </li>
-          {process.env.REACT_APP_HOLOFUEL_APP !== 'true' && <li>
+          {process.env.REACT_APP_HOLOFUEL_AS_HOLOCHAIN_APP !== 'true' && <li>
```
Comment (robbiecarlton) [inline src/holofuel/components/SideMenu/SideMenu.js]: This should use `REACT_APP_HOLOFUEL_APP` (see overall review comment)

File: src/root.js [hunk 5]
```diff
@@ -15,7 +15,7 @@ import { FlashMessageProvider as HFFlashMessageProvider } from 'holofuel/context
 import HPAdminRouter from './HPAdminRouter'
 
 export function App () {
-  if (process.env.REACT_APP_HOLOFUEL_APP) {
+  if (process.env.REACT_APP_HOLOFUEL_AS_HOLOCHAIN_APP === 'true') {
```
Comment (robbiecarlton) [inline src/root.js]: This should use `REACT_APP_HOLOFUEL_APP` (see overall review comment)

**Review by robbiecarlton — CHANGES_REQUESTED** [review]
This is definitely something we need to do. However, I think this implementation collapses two distinct questions that we want to keep separate:

1) Are we running JUST HoloFuel, or all of hp-admin?

2) Are we running inside of HPOS, or as a raw holochain app.

So I think we want two different env variables:

`REACT_APP_HOLOFUEL_APP` (which we already have)
and
`REACT_APP_RAW_HOLOCHAIN` (or something. I can't find a name I love for this var, if you have something better use that)

**Review by robbiecarlton — APPROVED** [review]

**Pull request merged**